### PR TITLE
Adds Visual Studio 2022 to the version selector of vstest

### PR DESCRIPTION
**Pull request by jessehouwing**
**Task name**: vstest

**Description**: 
Adds the explicit choice of Visual Studio 2022 (v17) to the version selector of the vstest@2 and vetest@3 tasks and extends the code to find the visual studio installer. 

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** 
 * #17090
 * https://github.com/jessehouwing/azure-pipelines-tasks-zips/issues/6
 * https://github.com/microsoft/azure-pipelines-tasks/pull/17460

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by jessehouwing** [comment]
@phanikmmsft For some reason the auto-assign task failed, tagging you manually.

File: Tasks/VsTestV2/task.loc.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
   "version": {
     "Major": 2,
     "Minor": 210,
```
Comment (kirill-ivlev) [inline Tasks/VsTestV2/task.loc.json:20]: Could you please bump the task to the current sprint? (should be 214.0)

**Review by kirill-ivlev — APPROVED** [review]
In general LGTM, but could you please update the task version

**Review by LiliaSabitova — APPROVED** [review]

**Comment by jessehouwing** [comment]
What version does it need to be? 214.0?

**Comment by jessehouwing** [comment]
Done

File: Tasks/VsTestV2/task.loc.json (hunk 1 above)
Comment (jessehouwing) [reply Tasks/VsTestV2/task.loc.json]: Done

**Comment by jessehouwing** [comment]
@LiliaSabitova @kirill-ivlev Merge?

**Comment by jessehouwing** [comment]
@geekzter

**Comment by jessehouwing** [comment]
Need to update to 216?

**Comment by geekzter** [comment]
@kirill-ivlev can you have a look?

**Comment by jessehouwing** [comment]
Related / Duplicate: https://github.com/microsoft/azure-pipelines-tasks/pull/17460

**Comment by galakt** [comment]
Indeed, our changes are similar, but
* You updated v3 as well
* We have different changes, e.g. in `Tasks/VsTestV2/inputparser.ts`. Curious about

**Comment by jessehouwing** [comment]
> Indeed, our changes are similar, but
> 
> * You updated v3 as well
> * We have different changes, e.g. in `Tasks/VsTestV2/inputparser.ts`. Curious about

I might have missed a few things. The task work in my scenario, but there may be others I'm not using in my tasks.

**Comment by galakt** [comment]
I have questions about original code in the tasks. Now I am waiting for answer from area owner, I pinged him internally. Probably, after all we will combine our changes and finally merge it

**Comment by jessehouwing** [comment]
@galakt it sure could use a refactoring. I pulled in your changes from your fork and merged them with mine. V2 and V3 in my branch should have the same changes now.

**Comment by galakt** [comment]
I want to notify you, that our changes were combined and finally merged in #17858

**Comment by jessehouwing** [comment]
Woot!

**Pull request closed without merging**